tests/cli/main.cpp

```diff
@@ -1409,3 +1410,184 @@ BOOST_FIXTURE_TEST_CASE( general_storage, cli_fixture )
       throw;
    }
 }
+
+//////
+// Template copied
+//////
+template<typename Object>
+unsigned_int member_index(string name) {
+   unsigned_int index;
+   fc::typelist::runtime::for_each(typename fc::reflector<Object>::native_members(), [&name, &index](auto t) mutable {
+      if (name == decltype(t)::type::get_name())
+      index = decltype(t)::type::index;
+   });
+   return index;
+}
+
+///////////////////////
+// Wallet RPC
+// Test sign_builder_transaction2 with an account (bob) that has received a custom authorization
+// to transfer funds from another account (alice)
+///////////////////////
+BOOST_FIXTURE_TEST_CASE(cli_use_authorized_transfer, cli_fixture) {
+   try {
+      //////
+      // Initialize the blockchain
+      //////
+      auto db = app1->chain_database();
+
+      account_object nathan_acct = con.wallet_api_ptr->get_account("nathan");
+      INVOKE(upgrade_nathan_account);
+
+      // Register Alice account
+      const auto alice_bki = con.wallet_api_ptr->suggest_brain_key();
+      con.wallet_api_ptr->register_account(
+              "alice", alice_bki.pub_key, alice_bki.pub_key, "nathan", "nathan", 0, true
+      );
+      const account_object &alice_acc = con.wallet_api_ptr->get_account("alice");
```

Comment:
`wallet::get_account` returns an account object, not a reference (unlike `database::get_account`).